modules/apps/27-interchain-accounts/controller/keeper/account.go

```diff
@@ -17,8 +17,19 @@ import (
 // - An error is returned if the port identifier is already in use. Gaining access to interchain accounts whose channels
 // have closed cannot be done with this function. A regular MsgChannelOpenInit must be used.
 func (k Keeper) RegisterInterchainAccount(ctx sdk.Context, connectionID, owner, version string) error {
-	_, err := k.registerInterchainAccount(ctx, connectionID, owner, version)
-	return err
+	channelID, err := k.registerInterchainAccount(ctx, connectionID, owner, version)
+	if err != nil {
+		return err
+	}
+
+	portID, err := icatypes.NewControllerPortID(owner)
+	if err != nil {
+		return err
+	}
+
+	k.SetMiddlewareEnabled(ctx, portID, channelID)
```

Comment:
Thinking about change the signature of the private func `registerInterchainAccount` and having this func and the msg server func generate the port ID.

Thoughts?

Reply:
I don't have a preference. Whatever you think is best